staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -206,6 +208,15 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	ch := s.Watching.ResultChan()
 	done := req.Context().Done()
 
+	embeddedEncodeFn := s.EmbeddedEncoder.Encode
+	if encoder, supportsAllocator := s.EmbeddedEncoder.(runtime.EncoderAllocator); supportsAllocator {
+		memoryAllocator := serializer.AllocatorPool.Get().(*serializer.Allocator)
+		defer serializer.AllocatorPool.Put(memoryAllocator)
```

Comment:
Should these two lines go into embeddedEncodeFn ?

Reply:
I don't think so. We don't want to allocate/deallocate on every call to the encoding method. Instead, we allocate the buffer for the entire watch session and release it when we close the connection.

Reply:
OK - that's fair point. Maybe just add a comment.